File: Java/src/tema9programacionorientadaaobjetos/ficheros/Ejercicio1.java
package tema9programacionorientadaaobjetos.ficheros;

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;

/**
 * <p>
 * Escribe un programa que guarde en un fichero con nombre primos.dat los
 * números primos que hay entre 1 y 500.
 * </p>
 * 
 * @author Francisco Javier Frías Serrano
 * @version 1.0
 */
public class Ejercicio1 {

  public static void main(String[] args) {

    String ruta = "C:\\Users\\d18frsej2\\eclipse-workspace\\PGNJava\\src\\tema9programacionorientadaaobjetos\\ficheros\\primos.dat";

    try {
      BufferedWriter escribir = new BufferedWriter(new FileWriter(ruta));

      boolean esPrimo;
      for (int i = 1; i < 501; i++) {
        esPrimo = true;
        for (int j = 2; j < i; j++) {
          if (i % j == 0) {
            esPrimo = false;
          }
        }
        if (esPrimo) {
          escribir.write(i + "\n");
        }
      }
      System.out.println("Archivo guardado con éxito");

      escribir.close();

    } catch (IOException e) {
      System.err.println("No se ha podido escribir el archivo");
    }
    
  }
}

File: Python/poo/ficheros/Ejercicio1.py
'''
 * <p>
 * Escribe un programa que guarde en un fichero con nombre primos.dat los
 * números primos que hay entre 1 y 500.
 * </p>

@author: Francisco Javier Frías Serrano
'''

# main



def esPrimo(num):
    """
    Esta función comprueba si un número es primo
    :param num:
    :return: devuelve True si el número pasado por prámetro es primo
    """
    if num <= 1:
        return False
    elif num == 2:
        return True
    else:
        for i in range(2, num):
            if num % i == 0:
                return False
        return True

escritura = open("primos.dat", "w")

for x in range(501):
    """
    Con este for compruebo los números primos del 1 al 500 y los añado al escritura antes creado
    """
    if(esPrimo(x) == True):
        escritura.writelines(str(x) + " ")

escritura.close()
